File: poc-kronos-frontend/src/model/diario/Diario.js
class Diario {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.procedimento = dados ? dados.procedimento : null;
    this.recurso = dados ? dados.recurso : null;
    this.observacao = dados ? dados.observacao : null;
    this.idAtividade = dados ? dados.idAtividade : null;
    this.idTipoPrograma = dados ? dados.idTipoPrograma : null;
    this.nomePrograma = dados ? dados.nomePrograma : null;
  }

  static validations(required, maxLength) {
    return {
      procedimento: { required, maxLength: maxLength(500) },
      recurso: { required, maxLength: maxLength(500) },
      observacao: { maxLength: maxLength(500) },
      idAtividade: { required },
      idTipoPrograma: { required }
    };
  }

  static validationMessages() {
    return {
      procedimento: {
        required: "Procedimento é obrigatório",
        maxLength: "Procedimento só aceita no máximo 500 caracteres"
      },
      recurso: {
        required: "Recurso é obrigatório",
        maxLength: "Recurso só aceita no máximo 500 caracteres"
      },
      observacao: {
        maxLength: "Observação só aceita no máximo 500 caracteres"
      },
      idAtividade: {
        required: "Atividade é obrigatória"
      },
      idTipoPrograma: {
        required: "Tipo Programa é obrigatório"
      }
    };
  }
}
export default Diario;

File: src/main/java/br/com/kronos/backend/aplicacao/diario/Diario.java

package br.com.kronos.backend.aplicacao.diario;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Diario {
    
    private long id;
    private String procedimento;
    private String recurso;
    private String observacao;
    private Long idAtividade;
    private Integer idTipoPrograma;

	@Builder
    public Diario(long id, String procedimento, String recurso, String observacao, Long idAtividade, Integer idTipoPrograma) {

        this.id = id;
        this.procedimento = exigirTamanho(procedimento, "Procedimento", 1, 500); 
        this.recurso = exigirTamanho(recurso, "Recurso", 1, 500); 
        this.observacao = exigirTamanho(observacao, "Observação", 0, 500); 
        this.idAtividade = exigirNaoNulo(idAtividade, "Atividade"); 
        this.idTipoPrograma = exigirNaoNulo(idTipoPrograma, "Tipo de programa"); 
	}
}